kotlintest-assertions/src/main/kotlin/io/kotlintest/matchers/throwable/matchers.kt

```diff
@@ -0,0 +1,63 @@
+package io.kotlintest.matchers.throwable
+
+import io.kotlintest.Matcher
+import io.kotlintest.Result
+import io.kotlintest.should
+import io.kotlintest.shouldNot
+
+infix fun Throwable.shouldHaveMessage(message: String) = this should haveMessage(message)
+infix fun Throwable.shouldNotHaveMessage(message: String) = this shouldNot haveMessage(message)
+fun haveMessage(message: String) = object : Matcher<Throwable> {
+  override fun test(value: Throwable) = Result(
+      value.message == message,
+      "Throwable should have message \"$message\", but instead got ${value.message}",
```

Comment:
I believe this should be treated in a special way. I have to look where it is, but we have a `transformString` of some type somewhere in our code.

I saw that in a recent commit from @sksamuel .

This `${value.message}` might become confusing if it's either null or empty

"Throwable should have message $message but instead got ."

The function I cited already handles this kind of case, I just need to remember which is it

Reply:
done!